pkg/runtime/framework/plugins/jobset/jobset.go

```diff
@@ -159,3 +164,52 @@ func (j *JobSet) TerminalCondition(ctx context.Context, trainJob *trainer.TrainJ
 	}
 	return nil, nil
 }
+
+func (j *JobSet) Validate(runtimeJobTemplate client.Object, runtimeInfo *runtime.Info, oldObj, newObj *trainer.TrainJob) (admission.Warnings, field.ErrorList) {
+
+	var allErrs field.ErrorList
+	specPath := field.NewPath("spec")
+	runtimeRefPath := specPath.Child("runtimeRef")
+
+	jobSet, ok := runtimeJobTemplate.(*jobsetv1alpha2.JobSet)
+	if !ok {
+		return nil, nil
+	}
+
```

Comment:
```suggestion
	rJobInitContainerNames := make(map[string]sets.Set[string])
	for _, rJob := range jobSet.Spec.ReplicatedJobs {
		rJobInitContainerNames[rJob.Name] = sets.New[string]()
		for _, c := range rJob.Template.Spec.Template.Spec.InitContainers {
			rJobInitContainerNames[rJob.Name].Insert(c.Name)
		}
	}
```

Could you create a set of rJob names so that we can reduce the iteration?